### PR TITLE
Add vexflow@1.2.84 w/ npm auto-update

### LINKED ISSUE #11542: [Request] Add Vexflow
**Library name:** Vexflow
**Git repository url:** https://github.com/0xfe/vexflow
**npm package name or url** (if there is one): https://www.npmjs.com/package/vexflow
**License (List them all if it's multiple):** MIT
**Official homepage:** http://www.vexflow.com/

### PR DESCRIPTION
**Pull request by qwe19272375**
close #11542, cc @0xfe

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/0xfe/vexflow
 * Official website (optional, not the repository): http://vexflow.com
 * NPM package url (optional): https://www.npmjs.com/package/vexflow
 * License and its reference: [MIT](https://github.com/0xfe/vexflow/blob/master/LICENSE)
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 105
   - Count of stars: 1880
   - Count of forks: 388
 * NPM download stats (optional):
   - Downloads in the last day: 19
   - Downloads in the last week: 222
   - Downloads in the last month: 1186

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/vexflow/1.2.84/vexflow-tests.js [hunk 1]
```diff
@@ -0,0 +1,13838 @@
+/**
+ * VexFlow 1.2.84 built on 2017-07-30.
+ * Copyright (c) 2010 Mohit Muthanna Cheppudira <mohit@muthanna.com>
+ *
+ * http://www.vexflow.com  http://github.com/0xfe/vexflow
+ *//**
+ * VexFlow Test Support Library
```
Comment (qwe19272375) [inline ajax/libs/vexflow/1.2.84/vexflow-tests.js]: @sufuf3 the comment say that this is vexflow test support library, so I think this can be included too?

**Comment by qwe19272375** [comment]
@cdnjs/intern3 please help me review this PR, thanks.

File: ajax/libs/vexflow/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    "tablature"
+  ],
+  "npmName": "vexflow",
+  "npmFileMap": [
+    {
+      "basePath": "releases",
+      "files": [
+        "vexflow*"
+      ]
+    }
```
Comment (extend1994) [inline ajax/libs/vexflow/package.json]: Please add the pattern to fetch the old versions v1.2.3 ~ v1.2.5 on npm. (You might need to check if we need all `vexflow*` under `lib`)
Like what https://github.com/cdnjs/cdnjs/pull/11439#issuecomment-312471744 did.

File: ajax/libs/vexflow/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "notation",
+    "guitar",
+    "tablature"
+  ],
+  "npmName": "vexflow",
+  "npmFileMap": [
+    {
+      "basePath": "releases",
+      "files": [
+        "vexflow*"
```
Comment (extend1994) [inline ajax/libs/vexflow/package.json]: Let's modify this to `**/vexflow!(*.html)`. `releases` is similar to `dist`. We usually need most files under them. The pattern I suggest is based on we need all `**/verflow*` except `docs/vexflow_font.html` in v1.2.6 ~ v1.2.83. If you have a better idea. Just do it!

**Comment by qwe19272375** [comment]
@cdnjs/intern3 please help me review this PR, thanks!

**Comment by qwe19272375** [comment]
@hare1039 @extend1994 please help me review this PR, thanks!

**Comment by sufuf3** [comment]
@qwe19272375 Would you fix the PR?
The author response your question.
Thanks.

**Comment by sufuf3** [comment]
ping @qwe19272375

**Comment by sashberd** [comment]
@extend1994 The PR was updated. FYR

**Pull request merged**